sharding-transaction/src/main/java/io/shardingsphere/transaction/manager/xa/AbstractXATransactionManager.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2016-2018 shardingsphere.io.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * </p>
+ */
+
+package io.shardingsphere.transaction.manager.xa;
+
+import io.shardingsphere.core.event.transaction.xa.XATransactionEvent;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.transaction.HeuristicMixedException;
+import javax.transaction.HeuristicRollbackException;
+import javax.transaction.NotSupportedException;
+import javax.transaction.RollbackException;
+import javax.transaction.SystemException;
+import javax.transaction.UserTransaction;
+import java.sql.SQLException;
+
+@Slf4j
+public abstract class AbstractXATransactionManager implements XATransactionManager {
```

Comment:
maybe AbstractXAtransactionManager only be used by AtomikosTransactionManager,
it think narayaTransactionManager was implemented in other repo and don't depend on SS,
so it seems that we should do init directly in AtomikosTransactionManager, and I think init() method can be removed.

Reply:
OK, it looks good to remove this abstract class.